evap/results/tests/test_views.py

```diff
@@ -203,13 +196,34 @@ def helper_test_answer_visibility(self, username, expect_page_not_visible_first=
         number_of_disabled_grade_badges = str(page).count("grade-bg-result-bar text-center grade-bg-disabled")
         self.assertEqual(number_of_disabled_grade_badges, 1)
 
-        # remove additional voter
-        self.course.voters.remove(self.student2)
-
     def test_answer_visibility(self):
-        self.helper_test_answer_visibility("staff")
-        self.helper_test_answer_visibility("responsible")
-        self.helper_test_answer_visibility("student", expect_page_not_visible_first=True)
+        self.let_user_vote_for_course(self.student1, self.course)
+        self.course.evaluation_end()
+        self.course.review_finished()
+        self.course.publish()
+        self.course.save()
+        self.assertEqual(self.course.voters.count(), 1)
+        caches["results"].clear()
+        self.helper_test_answer_visibility_one_voter("staff")
+        self.course = Course.objects.get(id=self.course.id)
+        self.helper_test_answer_visibility_one_voter("responsible")
+        self.helper_test_answer_visibility_one_voter("student", expect_page_not_visible=True)
+
+        # add second voter
+        self.course.unpublish()
+        self.course.reopen_evaluation()
+        self.course.save()
+        self.let_user_vote_for_course(self.student2, self.course)
```

Comment:
put this into a separate test? the unpublishing and cache clearing shouldn't be necessary then.

Reply:
cache clearing is indeed not necessary but we need some kind of reset for the course because "Modifications to in-memory objects from setup work done at the class level will persist between test methods."
so the course would still be in state `published` in the second test.

Reply:
yeah, that's why i suggested to put the second part into a separate test :)

Reply:
yeah, you probably need a `self.course = Course.objects.get(pk=self.course.pk)`, but you shouldn't need the unpublishing part.

Reply:
...and please put that into a setUp method, otherwise it will be dependent on the execution order.

Reply:
okay